### DISCUSSION TITLE
Change directory in GDL?

### DISCUSSION THREAD

**Discussion by tushara04** [gnudatalanguage/gdl#1356] [Q&A] [answered]
So, I have installed GDL Windows 1.0.1 version and I use the command line via WSL.
And, for that, I have set my gnudatalangage folder as the directory. But, say, I create a file inside gnudatalanguage directory and save a .pro file. When I try opening it, GDL says the directory or file does not exist. So, i realized that I have to choose absolute directory and hence I would have to move through my directories often, right?
So, how do I do that? Do I set my startup file as such (if so how? please, guide!)? And if not, then like as in ubuntu, we can do, cd GDL, cd new_folder, etc. So, how do we do the similar thing here on GDL?

(also what is cmake? I don't get it!)

Also, say, a file hello.pro is in a folder gdlisfun, but my directory is set to gdl (gdlisfun is inside gdl). So, how do I access hello.pro without having to go to the settings and change the directory and then change back again and all?

**Comment by slayoo** [comment, marked as answer]
Hi, 
- you can change the working directory of gdl by:
```
GDL> cd, 'new_folder'
```
- CMake is a tool needed to compile GDL from source. If you use a pre-compiled GDL package, CMake is not needed

**Reply by tushara04** [reply]
Right, got it. Thank you so much!

**Reply by tushara04** [reply]
Hi. 
I am sorry for not asking this doubt immediately but here it is:
so, I tried running 
`GDL> cd, Tests`
(Tests is a folder inside the directory that I have set in the settings)
but I get this:
<img width="274" alt="image" src="https://user-images.githubusercontent.com/108838006/183849975-13a60b42-d26a-4f74-8780-73d13873b499.png">

Now, you might say I am checking the wrong folder. 
But, see, when I ran the test2.pro file, which is located in the folder gnudatalanguage (the directory which I have set in the settings)
`GDL> .run test2`
it worked: 
<img width="647" alt="image" src="https://user-images.githubusercontent.com/108838006/183850374-cceb18cd-4d3b-44f0-bbac-519ede0a5b97.png">

but the test1.pro file, which is present inside a folder Tests (which is inside the gnudatalanguage folder), it doesn't work. That should mean that my directory is set to gnudatalanguage.
And that's true as in the settings: 
<img width="719" alt="image" src="https://user-images.githubusercontent.com/108838006/183850717-9dc73560-c50e-41d8-9053-b5ffbb5f73b5.png">

But, when I type
`GDL> cd, !dir`

I get this: 
<img width="554" alt="image" src="https://user-images.githubusercontent.com/108838006/183851169-c98a9ec2-28e5-4c9b-83f6-b9b14c9fa486.png">

What is happening here? I don't understand. Why doesn't it say that my directory is "D:/Program Files/GDL/gnudatalanguage"? 
What I want to do is that I want the directory that I have set in the settings as default and then use "cd" to access folders (such as Tests folder) inside the directory. How do i do that?

**Comment by GillesDuvert** [comment]
text (like directory names) in GDL must be written in double quotes, so it is:
`cd,"Tests"`
This is quite standard for all languages, interpreted as GDL or compiled as C. The program must know if '`test`' is an array of characters (`"test`") or a variable or expression (`test`) it has to find/modify/decipher .

**Reply by tushara04** [reply]
eyy, it worked! Thanks so much!
Just one final thing, hopefully. How do I go back now? Like, say, I did `cd, "Tests"` and now I want to go back to the earlier dir. How do I do that without having to restart the terminal?

**Reply by GillesDuvert** [reply]
have a look at [this](https://earth.bu.edu/public/amerb/IDL_Intro/IDL_Intro_12March2014.pdf) which is very complete. Search for 'cd,' .
Just change mentally IDL by GDL. Some rare IDL commands or syntax will not be available in GDL, but normally GDL now covers a very large amount of useful routines. All the fancy "workbench" environment in the documentation may even work, see https://github.com/gnudatalanguage/gdlde (separate development). Also, IDL examples files and procedures are copyrighted and absent from GDL.

**Reply by tushara04** [reply]
Wow, the guide!! That's exactly what I wanted all this time! Thanks SO much! <3
And, yep, got everything you said! thanks!! :)
